core/src/main/java/hudson/Functions.java

```diff
@@ -851,6 +851,10 @@ public static String getFooterURL() {
         return JobPropertyDescriptor.getPropertyDescriptors(clazz);
     }
 
+    public static List<JobPropertyDescriptor> getJobPropertyDescriptors(Job job) {
+        return DescriptorVisibilityFilter.apply(job, JobPropertyDescriptor.getPropertyDescriptors(job.getClass()));
+    }
```

Comment:
@KostyaSha right, it still makes sense (to me) to put the method with a class to which it has a logical connection. The fact that a UI binding was the first thing that made a requirement for the method seems secondary to me. Anyway ... nothing we need to debate I think ... I was just wondering.